File: src/main/java/org/jdc/template/ux/individualedit/IndividualEditContract.java
package org.jdc.template.ux.individualedit;

import org.jdc.template.model.database.main.individual.Individual;
import org.threeten.bp.LocalDate;
import org.threeten.bp.LocalTime;

public class IndividualEditContract {
    interface View {
        void showIndividual(Individual individual);
        boolean validateIndividualData();
        void getIndividualDataFromUi(Individual individual);
        void close();
        void showAlarmTimeSelector(LocalTime time);
        void showAlarmTime(LocalTime time);
        void showBirthDateSelector(LocalDate date);
        void showBirthDate(LocalDate date);
    }

    public class Extras {
        public static final String EXTRA_ID = "INDIVIDUAL_ID";
    }
}

File: src/main/kotlin/org/jdc/template/ux/individualedit/IndividualEditContract.kt
package org.jdc.template.ux.individualedit

import android.content.Intent
import me.eugeniomarletti.extras.ActivityCompanion
import me.eugeniomarletti.extras.intent.IntentExtra
import me.eugeniomarletti.extras.intent.base.Long
import org.jdc.template.model.database.main.individual.Individual
import org.threeten.bp.LocalDate
import org.threeten.bp.LocalTime

class IndividualEditContract {
    interface View {
        fun showIndividual(individual: Individual)
        fun validateIndividualData(): Boolean
        fun getIndividualDataFromUi(individual: Individual)
        fun close()
        fun showAlarmTimeSelector(time: LocalTime)
        fun showAlarmTime(time: LocalTime)
        fun showBirthDateSelector(date: LocalDate)
        fun showBirthDate(date: LocalDate)
    }

    companion object : ActivityCompanion<IntentOptions>(IntentOptions, IndividualEditActivity::class)

    object IntentOptions {
        var Intent.individualId by IntentExtra.Long(defaultValue = -1L)
    }
}
